packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -238,6 +238,7 @@ export interface SdkBodyModelPropertyType extends SdkModelPropertyTypeBase {
   serializedName: string;
   isMultipartFileInput: boolean;
   visibility?: Visibility[];
+  flatten: boolean;
```

Comment:
Adding this could potentially break emitters if they are building the SdkBodyModelPropertyType. Can we mark this as optional?

Reply:
Emitters shouldn't be building body model property types, the structure of the tcgc types has been that tcgc owns the structure of the interfaces, and emitters should either only access the properties on the sdk type